### PR TITLE
fix: running script with `--broadcast` for a transaction sequence can error out due to nonce desync from rpc latency

### PR DESCRIPTION
**Pull request by pogobounce**
## Motivation

On some chains with fast block times and/or relatively high latency providers, running a deployment script with `--broadcast` that sends a sequence of transactions can lead to the script failing, as the RPC will report an older nonce for the EOA.

## Solution

If the RPC is out of sync with the current nonce, there's two options:

If it returns a nonce that is ahead, it likely means that the EOA executed some other transaction meanwhile, so the script should stop. 

However, if the nonce is ahead, the culprit is likely a slow RPC. In that case, it makes sense to continue to proceed with the transaction sequence.

Treating these two cases separately means the sequence of transactions won't get stopped out.

### CODE REVIEW

**Review by klkvr — CHANGES_REQUESTED** [review]
provider reporting nonce which is behind might mean several things — there might've been a reorg or node might just still be updating state even though block with previous nonce was already included but `latest` state is not yet updated

in both cases we'd want to wait until we are sure that previous transaction is available in state because otherwise estimate_gas below might fail

let's add a loop here which would do several attempts at fetching and comparing nonce with some delay, and if after eg 5 iterations we still have a mismatch we'd always fail

File: crates/script/src/broadcast.rs [hunk 1]
```diff
@@ -67,11 +67,24 @@ pub async fn send_transaction(
...
+                match nonce.cmp(&tx_nonce) {
+                    Ordering::Greater => {
+                        bail!("EOA nonce changed unexpectedly while sending transactions. Expected {tx_nonce} got {nonce} from provider.")
+                    }
+                    Ordering::Less => {
+                        if attempt == 4 {
+                            bail!("After 5 attempts, provider nonce ({nonce}) is still behind expected nonce ({tx_nonce}).")
+                        }
+                        warn!("Expected nonce ({tx_nonce}) is ahead of provider nonce ({nonce}). Retrying in 1 second...");
+                        std::thread::sleep(std::time::Duration::from_millis(1000));
```
Comment (klkvr) [inline crates/script/src/broadcast.rs:82]: this should be `tokio::time::sleep`

File: crates/script/src/broadcast.rs [hunk 2]
```diff
@@ -67,11 +67,24 @@ pub async fn send_transaction(
...
+                    }
+                    Ordering::Equal => {
+                        // Nonces are equal, we can proceed
```
Comment (klkvr) [inline crates/script/src/broadcast.rs:85]: shouldn't we break here?

**Review by klkvr — CHANGES_REQUESTED** [review]
nice, a couple nits

File: crates/script/src/broadcast.rs (hunk 2 above)
Comment (pogobounce) [reply crates/script/src/broadcast.rs:85]: you're right, changed

**Review by klkvr — APPROVED** [review]

**Pull request merged**